cpp/src/dbscan/dbscan.cuh

```diff
@@ -66,30 +67,31 @@ size_t compute_batch_size(size_t& estimated_memory,
   // Limit batch size to number of owned rows
   batch_size = std::min((size_t)n_owned_rows, batch_size);
 
-  // To avoid overflow, we need: batch_size <= MAX_LABEL / n_rows (floor div)
-  Index_ MAX_LABEL = std::numeric_limits<Index_>::max();
-  if (batch_size > static_cast<std::size_t>(MAX_LABEL / n_rows)) {
-    Index_ new_batch_size = MAX_LABEL / n_rows;
-    CUML_LOG_WARN(
-      "Batch size limited by the chosen integer type (%d bytes). %d -> %d. "
-      "Using the larger integer type might result in better performance",
-      (int)sizeof(Index_),
-      (int)batch_size,
-      (int)new_batch_size);
-    batch_size = new_batch_size;
+  if (eps_nn_method != EpsNnMethod::RBC) {
+    // To avoid overflow, we need: batch_size <= MAX_LABEL / n_rows (floor div)
+    Index_ MAX_LABEL = std::numeric_limits<Index_>::max();
+    if (batch_size > static_cast<std::size_t>(MAX_LABEL / n_rows)) {
+      Index_ new_batch_size = MAX_LABEL / n_rows;
+      CUML_LOG_WARN(
+        "Batch size limited by the chosen integer type (%d bytes). %d -> %d. "
+        "Using the larger integer type might result in better performance",
+        (int)sizeof(Index_),
+        (int)batch_size,
+        (int)new_batch_size);
+      batch_size = new_batch_size;
+    }
+
+    // Warn when a smaller index type could be used
+    if ((sizeof(Index_) > sizeof(int)) &&
+        (batch_size < std::numeric_limits<int>::max() / static_cast<std::size_t>(n_rows))) {
+      CUML_LOG_WARN(
```

Comment:
Small existential question: are these two logs not more of the info as opposed to warning level? My reasoning is that they are not indicating a potential problem but just lower performance.

I'm fine if we keep them as warnings, but just got me thinking.

Reply:
Hmmm I agree, this should probably be `INFO`. More general, meta thought - I don't think `int64` indexes are as bad performance wise as `float64` though as most of our computation bound kernels operate in the floating point space, and the integers are generally just predictive labels.